pkg/simpleschema/transform.go

```
@@ -205,15 +208,46 @@ func (tf *transformer) applyMarkers(schema *extv1.JSONSchemaProps, markers []*Ma
 		case MarkerTypeDescription:
 			schema.Description = marker.Value
 		case MarkerTypeMinimum:
-			if val, err := strconv.ParseFloat(marker.Value, 64); err == nil {
+			if val, err := strconv.ParseFloat(marker.Value, 64); err != nil {
+				return fmt.Errorf("failed to parse minimum enum value: %w", err)
+			} else {
 				schema.Minimum = &val
 			}
 		case MarkerTypeMaximum:
-			if val, err := strconv.ParseFloat(marker.Value, 64); err == nil {
+			if val, err := strconv.ParseFloat(marker.Value, 64); err != nil {
+				return fmt.Errorf("failed to parse maximum enum value: %w", err)
+			} else {
 				schema.Maximum = &val
 			}
+		case MarkerTypeEnum:
+			enumValues := strings.Split(marker.Value, ",")
+
+			var enumJSONValues []extv1.JSON
+
+			for _, val := range enumValues {
+				val = strings.TrimSpace(val)
+				var rawValue []byte
```

Not sure how is this handling invalid enums such as `a,b,,,,c` - and i'm not quiet grokking how are unit tests failing here. Could be something wrong with the assert statements

---

There's really a great edge case as in this case `string.Split(marker.Value, ",")` will make it `["a", "b", "", "", "", "c"]` and later loop and code further make this an empty string which is wrong. As in the case of integers it fails early when it hits the empty string and my test case checks the string that's why its failing. Let me add a if statement and a test case for this.

---

Good catch, indeed `strconv.ParseInt` errors with empty strings